pkg/tempo/servicegraphprocessor/factory.go

```diff
@@ -0,0 +1,53 @@
+package servicegraphprocessor
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+)
+
+const (
+	// TypeStr is the unique identifier for the Prometheus service graph exporter.
+	TypeStr = "service_graphs"
+
+	defaultWait     = time.Second * 10
+	defaultMaxEdges = 10_000
```

Comment:
I have some operational concerns of storing 10k edge objects in the store at any given time, even though our current scale of operation is much higher (something like 70k spans/sec through each agent pod?). We might end up either blowing up the agent memory _or_ having inaccurate graphs because we are unable to process all spans. Thoughts?

Reply:
It's difficult to say. The objects stored are way smaller that the traces that are naturally in memory in the pipeline. If you consider that tail sampling might be buffering a similar amount of entire traces, it doesn't sound that bad.

Also, most items should be very short lived. If the edge is _complete_, it's immediately removed.

My initial approach was based on processing entire traces by extracting the grouping by trace functionality out of tail sampling (see https://github.com/grafana/agent/pull/743), so both features leveraged grouping only in one place. The main drawback is that you have to buffer traces for at least 1s in the sampling processor anyway.

If the deprecation of the tail sampling processor moves forward (see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1797), that approach is the winner for me, since we won't have to buffer in the processor anymore. But it's not the case at the moment.